### PR TITLE
docs: clarify smooth l1 == l1 when beta == 0

### LINKED ISSUE #68558: [docs] SmoothL1Loss to document behavior at beta=0
https://pytorch.org/docs/stable/generated/torch.nn.SmoothL1Loss.html#torch.nn.SmoothL1Loss:

The docs only specify what happens when beta tends to zero (beta -> 0), it also specifies that beta must be non-negative.

What happens when beta = 0? Does it compute L1 loss in this case?

I think it would be useful to cover this case explicitly in the docs

cc @brianjo @mruberry @albanD @jbschlosser @walterddr

### PR DESCRIPTION
**Pull request by jaketae**
Fixes https://github.com/pytorch/pytorch/issues/68558.

### CODE REVIEW

**Review by jbschlosser — APPROVED** [review]
LGTM thanks!

**Review by Neilblaze — APPROVED** [review]
